dvc/repo/add.py

```diff
@@ -38,6 +41,20 @@ def add(
         raise RecursiveAddingWhileUsingFilename()
 
     targets = ensure_list(targets)
+
+    if output:
+        try:
+            dname = os.path.dirname(output)
+
+            if dname and not os.path.isdir(dname):
+                os.makedirs(dname)
+
+            shutil.copy2(targets[0], output)
```

Comment:
This will be really slow with big files/dirs. We could be able to add it without this uneeded copy.

Reply:
Also, you are checking that there is only one target in CLI, but not in API (they could be used separately), so `targets[0]` needs some reassurance.